website_snippet_barcode/templates/snippet_template.xml

```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2017 LasLabs Inc.
+     License LGPL-3 or later (http://www.gnu.org/licenses/lgpl.html). -->
+
+<odoo>
+    <template id="snippets" inherit_id="website.snippets">
+        <xpath expr="//div[@id='snippet_content']/div[@class='o_panel_body']"
+               position="inside">
+            <t t-snippet="website_snippet_barcode.barcode_snippet"
+               t-thumbnail="/website_snippet_barcode/static/description/snippet_icon.png"
+               />
+        </xpath>
+
+        <xpath expr="//div[@id='snippet_options']" position="inside">
+            <div data-js="barcode"
+                 data-selector=".s_barcode"
+                 data-drop-near="p, h1, h2, h3, blockquote, .well, .s_panel, .oe_share"
+                 data-drop-in=".content, .row">
+                <li class="dropdown-submenu">
+                    <a tabindex="-1" href="#"><i class="fa fa-barcode"/>Barcode Type</a>
+                    <ul class="dropdown-menu" name="barcode-type">
+                        <t t-foreach="['Codabar', 'Code11', 'Code128', 'EAN13', 'EAN8',
+                                       'Extended39', 'Extended93', 'FIM', 'I2of5', 'MSI', 'POSTNET',
+                                       'QR', 'Standard39', 'Standard93', 'UPCA', 'USPS_4State']"
+                           t-as="i">
+                            <li t-att-data-select_type='i'>
+                                <a href="#" t-esc="i" />
+                            </li>
+                        </t>
+                    </ul>
+                </li>
+
+                <li class="dropdown-submenu">
+                    <a tabindex="-1" href="#"><i class="fa fa-link"/>Barcode Value</a>
+                    <ul class="dropdown-menu" name="barcode-value">
+                        <li data-select_value="custom"><a href="#">Set Custom Value...</a></li>
+                        <li class="divider"></li>
+                        <li data-select_value="current"><a href="#">Current Page URI</a></li>
+                    </ul>
+                </li>
+
+                <li class="dropdown-submenu">
+                    <a tabindex="-1" href="#"><i class="fa fa-picture-o"/>Aspect Ratio</a>
+                    <ul class="dropdown-menu" name="barcode-aspectratio">
+                        <t t-foreach="range(1, 6)" t-as="i">
+                            <li t-att-data-select_aspectratio='i'>
+                                <a href="#" t-esc='"1:%s" % i' />
+                            </li>
+                        </t>
+                    </ul>
+                </li>
+
+                <li data-select_humanreadable="true">
+                    <a href="#"><i class="fa fa-user"/>Human Readable</a>
+                </li>
+            </div>
+        </xpath>
+    </template>
+
+    <template id="barcode_snippet" name="Barcode">
+        <div class="s_barcode col-md-12">
```

Comment:
Take a look at the current "Image Floating" snippet. We're going for something like that, vs the full-screen block that we have here.

```
<div class="o_image_floating o_margin_l pull-right s_image_floating">
        <div class="o_container">
            <img class="img img-rounded img-responsive" src="/web/image/website.s_image_floating_default_image" alt="Odoo CMS- Sample image floating" data-original-title="" title="">
        </div>
        <div class="o_footer">
            <small class="text-muted o_default_snippet_text">A great way to catch your reader's attention is to tell a story. Everything you consider writing can be told as a story.</small>
        </div>
    </div>
```

The top part would be the barcode, bottom would be the barcode numbering/text if the option is on.

Reply:
Works for me. A couple related questions:

* If I add the footer, should I remove the option of generating the barcode image with the human-readable portion? Or make it an option (in-image vs footer text)?

* I added the option to use it as a block (e.g. replacing one of the image/text blocks in the "Three Columns" snippet) because the "Image Floating" placement/sizing options seemed overly restrictive. Should I keep that or no? It does making resizing a little more complicated (as you noted in another comment).

Reply:
We would probably just optionally show/hide the footer. I think the option is good.

Regarding block - I'm not sure how useful a block barcode will be. I've never personally seen a barcode spanning the width of an entire page, and I'm not sure what elements we could combine it with that would fit all use cases.

What restrictions were you noticing with floating image?

Reply:
A full-width barcode would be pretty silly, definitely. I was placing it in the more structured snippets like three columns, references, features, and a couple others. Not sure how often that's likely to be an appropriate place for a barcode, though.

It's possible that the floating image placement just seemed restrictive when working with a bare-bones demo page without much content. I'm happy to remove the block option and have it behave like a floating image, if it works well in real-life use.